torch/_export/serde/serialize.py

```diff
@@ -1876,33 +1899,23 @@ def deserialize_graph(self, serialized_graph: Graph) -> torch.fx.Graph:
         # Handle the tensor metas.
         for name, tensor_value in serialized_graph.tensor_values.items():
             log.debug("[deserialize_tensor_meta] %s (input): %s", name, tensor_value)
-            meta_val = self.deserialize_tensor_meta(tensor_value)
-            log.debug("[deserialize_tensor_meta] %s (output): %s", name, meta_val)
-            self.serialized_name_to_meta[name] = meta_val
+            self.serialized_name_to_meta[name] = lambda v=tensor_value: self.deserialize_tensor_meta(v)
```

Comment:
Have you checked Python semantics re: capture of `tensor_value` here? Hope it doesn't pick up values assigned later and instead does what you think it does...which is capture the value in *this* iteration?